src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrSyncToolingE2ETest.cs

```diff
@@ -0,0 +1,624 @@
+    }
+
+    [TearDown]
+    public void DeleteCurrentTestDirectory()
+    {
+        try
+        {
+            if (_currentTestDirectory is not null)
+            {
+                DeleteDirectory(_currentTestDirectory.ToString());
+            }
+        }
+        catch
+        {
+            // Ignore
+        }
+    }
+
+    [Test]
+    public async Task RepoIsInitializedTest()
+    {
+        var testRepoFilePath = _vmrPath / "src" / "test-repo" / "test-repo-file.txt";
+        var dependencyFilePath = _vmrPath / "src" / "dependency" / "dependencyFile.txt";
+
+        var commit = await GetRepoLastCommit(_privateRepoPath);
+        await CallDarcInitialize(_vmrPath, _tmpPath, "test-repo", commit);
+
+        var expectedFilesFromRepos = new List<LocalPath> 
+        {
+            testRepoFilePath, 
+            dependencyFilePath
+        };
+
+        var expectedFiles = GetExpectedFilesInVmr(
+            _vmrPath,
+            new[] { "test-repo", "dependency" },
+            expectedFilesFromRepos
+        );
+        
+        CheckDirectoryContents(_vmrPath, expectedFiles);
+        CheckFileContents(testRepoFilePath, "Test repo file");
+        CheckFileContents(dependencyFilePath, "File in the dependency repo");
+        await CheckAllIsCommited(_vmrPath);
+    }
+
+    [Test]
+    public async Task FileChangesAreSyncedTest()
+    {
+        var testRepoFilePath = _vmrPath / "src" / "test-repo" / "test-repo-file.txt";
+        var dependencyFilePath = _vmrPath / "src" / "dependency" / "dependencyFile.txt";
+
+        await RepoIsInitializedTest();
+
+        File.WriteAllText(_privateRepoPath / "test-repo-file.txt", "Test changes in repo file");
+        await CommitAll(_privateRepoPath, "Changing a file in the repo");
+
+        var commit = await GetRepoLastCommit(_privateRepoPath);
+        await CallDarcUpdate(_vmrPath, _tmpPath, "test-repo", commit);
+
+        var expectedFilesFromRepos = new List<LocalPath>
+        {
+            testRepoFilePath,
+            dependencyFilePath
+        };
+
+        var expectedFiles = GetExpectedFilesInVmr(
+            _vmrPath,
+            new[] { "test-repo", "dependency" },
+            expectedFilesFromRepos
+        );
+
+        CheckDirectoryContents(_vmrPath, expectedFiles);
+        CheckFileContents(testRepoFilePath, "Test changes in repo file");
+        CheckFileContents(dependencyFilePath, "File in the dependency repo");
+        await CheckAllIsCommited(_vmrPath);
+    }
+
+    [Test]
+    public async Task FileIsIncludedTest()
+    {
+        await RepoIsInitializedTest();
+
+        File.Move(_privateRepoPath / "excluded" / "excluded.txt", _privateRepoPath / "excluded.txt");
+        await CommitAll(_privateRepoPath, "Move a file from excluded to included folder");
+        
+        var commit = await GetRepoLastCommit(_privateRepoPath);
+        await CallDarcUpdate(_vmrPath, _tmpPath, "test-repo", commit);
```

Comment:
I am thinking - is it needed to keep passing VMR/TMP path if it's always the same? Maybe the method could use these directly?